.gitignore

```diff
@@ -0,0 +1,14 @@
+cmake-build-debug*
+*build*
+spconfig.*
+*~
+.DS_Store
+*.swp
+.idea
+*.srctrl*
+
+# HPC coding conventions
+.clang-format
+.clang-tidy
+.cmake-format.yaml
+.pre-commit-config.yaml
```

Comment:
Did you intend to add this file?  Though most seem generically useful to ignore, one seems somewhat personal, e.g. ```.idea``` .  Some seem like one would like to ```git clean -d -f``` away.

Reply:
`.idea` is folder created by editors like IntelliJ and hence it's ignored. To remove everything including ignored files/folder, one can just add `-x` flag to `git clean`

Reply:
For things personal included by your IDE, you should modify the .gitignore in your home

Reply:
fixed